### PR TITLE
Fix `qml.gradients` docs examples

### PR DESCRIPTION
**Pull request by dwierichs**
- [x] Fix last open bug or open issue

**Context:**
Docs get out of sync over time.
In particular, the output shape of gradient transforms applied to QNodes was changed in #4945 but not updated in the docs.

**Description of the Change:**
Update examples in docs of `qml.gradients` module.

Also updates the `qml.kernels` docs, with a few very small changes.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/gradients/finite_difference.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def finite_diff(
     >>> params = jax.numpy.array([0.1, 0.2, 0.3])
     >>> jax.jacobian(circuit)(params)
     (Array([-0.38751727, -0.18884793, -0.3835571 ], dtype=float32),
-    Array([0.6991687 , 0.34072432, 0.6920237 ], dtype=float32))
+     Array([0.6991687 , 0.34072432, 0.6920237 ], dtype=float32))
```
Comment (Qottmann) [inline pennylane/gradients/finite_difference.py:283]:
```suggestion
    Array([0.6991687 , 0.34072432, 0.6920237 ], dtype=float32)

```

File: pennylane/gradients/finite_difference.py [hunk 2]
```diff
@@ -317,9 +313,9 @@ def finite_diff(
         >>> gradient_tapes, fn = qml.gradients.finite_diff(tape)
         >>> gradient_tapes
         [<QuantumTape: wires=[0], params=3>,
```
Comment (Qottmann) [inline pennylane/gradients/finite_difference.py:315]: This first one is still displayed as a QuantumTape?

File: pennylane/gradients/finite_difference.py [hunk 3]
```diff
@@ -338,32 +334,31 @@ def finite_diff(
...
-         tensor(0.93203912, requires_grad=True)))
+          tensor(-0.56464251, requires_grad=True),
+          tensor(-0.56464251, requires_grad=True)),
+         (tensor(0.93203912, requires_grad=True),
+          tensor(0.93203912, requires_grad=True),
+          tensor(0.93203912, requires_grad=True)))
 
         This gradient transform is compatible with devices that use shot vectors for execution.
 
+        >>> np.random.seed(251)
```
Comment (Qottmann) [inline pennylane/gradients/finite_difference.py:348]: wouldnt keep that in the docs no?

**Review by dwierichs — COMMENTED** [review]
Hope these comments help with reviewing swiftly.

File: pennylane/gradients/general_shift_rules.py [hunk 4]
```diff
@@ -366,11 +366,12 @@ def generate_multi_shift_rule(frequencies, shifts=None, orders=None):
 
     This corresponds to the gradient rule
 
+    TODO THERE IS A BUG CHECK RENDERING
     .. math::
 
         \frac{\partial^2 f}{\partial x\partial y} &= \frac{1}{4}
         \left[f(x+\pi/2, y+\pi/2) - f(x+\pi/2, y-\pi/2)\\
-        &~~~- f(x-\pi/2, y+\pi/2) + f(x-\pi/2, y-\pi/2) \right].
+        &- f(x-\pi/2, y+\pi/2) + f(x-\pi/2, y-\pi/2) \right].
```
Comment (Qottmann) [inline pennylane/gradients/general_shift_rules.py:374]: You should be able to use a latex align environment in ReST math:

```
.. math::

    \begin{align*}
    \beta_k & = \frac{2i\gamma}{2n+1}\sum_{j=-n}^n j \omega_k^j
    \end{align*}
```

(there are plenty of demos doing that)

**Review by Qottmann — APPROVED** [review]
Great update @dwierichs !
Looks good to me, just two minor things

File: pennylane/gradients/finite_difference.py (hunk 1 above)
Comment (Qottmann) [reply pennylane/gradients/finite_difference.py:283]: nevermind this one, didnt see the line above is still intact

File: pennylane/gradients/finite_difference.py (hunk 2 above)
Comment (dwierichs) [reply pennylane/gradients/finite_difference.py:315]: Yes, I think because it is the unshifted original tape, whereas the others are created newly :grimacing:

Comment (dwierichs) [reply pennylane/gradients/finite_difference.py:315]: Opened a discussion on this [here](https://github.com/PennyLaneAI/pennylane/pull/5596#discussion_r1584661194) (Sorry, I think I was too slow with posting these comments.

File: pennylane/gradients/general_shift_rules.py (hunk 4 above)
Comment (dwierichs) [reply pennylane/gradients/general_shift_rules.py:374]: Thanks! Yeah I'm a bit confused why this one does not render either way :thinking: Will figure it out!

File: pennylane/gradients/hadamard_gradient.py [hunk 5]
```diff
@@ -113,17 +113,18 @@ def hadamard_grad(
...
     >>> import jax
-    >>> dev = qml.device("default.qubit", wires=2)
+    >>> dev = qml.device("default.qubit")
     >>> @qml.qnode(dev, interface="jax", diff_method="hadamard")
     ... def circuit(params):
     ...     qml.RX(params[0], wires=0)
     ...     qml.RY(params[1], wires=0)
     ...     qml.RX(params[2], wires=0)
-    ...     return qml.expval(qml.Z(0)), qml.var(qml.Z(0))
+    ...     return qml.expval(qml.Z(0)), qml.probs(wires=0)
```
Comment (albi3ro) [inline pennylane/gradients/hadamard_gradient.py:122]: Why no variance?

File: pennylane/gradients/parameter_shift.py [hunk 6]
```diff
@@ -1049,11 +1049,14 @@ def param_shift(
...
-        (array(-0.3875172), array(-0.18884787), array(-0.38355704))
+        (tensor(-0.3875172, requires_grad=True),
+         tensor(-0.18884787, requires_grad=True),
+         tensor(-0.38355704, requires_grad=True))
 
         An advantage of using ``broadcast=True`` is a speedup:
 
-        >>> @qml.qnode(dev)
+        >>> import timeit
+        >>> @qml.qnode(qml.device("default.qubit"))
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/gradients/parameter_shift.py:1058]: did not work?

**Review by albi3ro — APPROVED** [review]
Thanks for doing this pass 🎉

File: pennylane/gradients/parameter_shift.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dwierichs) [reply pennylane/gradients/parameter_shift.py:1058]: Nope. Will investigate.

File: pennylane/gradients/hadamard_gradient.py (hunk 5 above)
Comment (dwierichs) [reply pennylane/gradients/hadamard_gradient.py:122]: It's explicitly not supported, see [here](https://github.com/PennyLaneAI/pennylane/blob/5cc8a622cbb48c6826dd4012730e7f94360dc491/pennylane/gradients/hadamard_gradient.py#L236) :sweat_smile:

File: pennylane/gradients/finite_difference.py [hunk 7]
```diff
@@ -317,9 +313,9 @@ def finite_diff(
         >>> gradient_tapes, fn = qml.gradients.finite_diff(tape)
         >>> gradient_tapes
         [<QuantumTape: wires=[0], params=3>,
-         <QuantumTape: wires=[0], params=3>,
-         <QuantumTape: wires=[0], params=3>,
-         <QuantumTape: wires=[0], params=3>]
+         <QuantumScript: wires=[0], params=3>,
+         <QuantumScript: wires=[0], params=3>,
+         <QuantumScript: wires=[0], params=3>]
```
[the remark this answers is not in the document]
Comment (trbromley) [reply pennylane/gradients/finite_difference.py:318]: I don't think it's technically inconsistent since `QuantumTape` is a child of `QuantumScript`, but yes does it look bad/confusing? :thinking: Overall, I think it's fine for now, let's see if someone is confused!

File: pennylane/gradients/parameter_shift.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dwierichs) [reply pennylane/gradients/parameter_shift.py:1058]: For tracking: #5598
